### PR TITLE
Chat logic

### PR DESCRIPTION
**Pull request by kobiez**
Configure the message.js to return the messages as a string.
I want to delete line 16 from the code, I think that we need to show the messages only in the GET request.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# CHAT APP
+
+*CHAT APP* , as the title suggests, will allow you to contact with other people via the platform.
+
+### Installation
+
+To install the app, you need to clone the directory of the code and open it with your code editor.
+
+### How To use
```
Comment (matan-co) [inline README.md:9]: In most of the systems you should tell something like
npm install
npm start.

And it should be with ``` section. Please take a look in the link that i sent you before.

**Review by matan-co — CHANGES_REQUESTED** [review]
Ill add more comments later

File: client/public/index.html [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Chat</title>
+    <style>
+        #allMessages {
+            border: 1px solid grey;
+            margin: 20px 20px 20px 0;
+            width: 275px;
+            height: 200px;
+        }
```
Comment (matan-co) [inline client/public/index.html:14]: move this logic to css file

File: client/public/index.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            "Content-Type": "application/json",
+        },
+    }).then((response) => response.json()).then((data) => {
+        console.log(data.user);
+        userId = data.user
+        document.getElementById("logBox").innerHTML = data.user;
+    }).catch((err) => {
+        console.error("Error: ", err);
+    })
+} */
```
Comment (matan-co) [inline client/public/index.js:66]: remove this code if not needed

File: client/public/index.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
+
+let userId;
+
+async function generateID() {
```
Comment (matan-co) [inline client/public/index.js:4]: Break this file into 2 files. authService and messageService. like we did in the server we are doing it in the client. 
Create a directory that call services. and put the logic of the api calls in there.

File: server/api/v1/auth/auth.test.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+const supertest = require("supertest");
+const httpStatus = require("http-status");
+const authLog = require("../../../index");
+const request = supertest(authLog);
+
+
+it("POST Auth - should create a new user id ", async () => {
+    const UserId = await request.post("/api/v1/auth").expect(httpStatus.OK);
```
Comment (matan-co) [inline server/api/v1/auth/auth.test.js:8]: UserId should be camel case.

File: server/api/v1/auth/auth.test.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    expect(UserId.body).toBeDefined();
+    console.log(UserId.body);
```
Comment (matan-co) [inline server/api/v1/auth/auth.test.js:10]: remove unneeded console.log

File: server/api/v1/message/message.js [hunk 7]
```diff
@@ -6,17 +6,17 @@ const router = express.Router();
 const messages = [];
 
 // Add chat api routes
-router.get("/", (req, res, next) => {
-  res.send(messages);
-});
 
 router.post("/", validateMessage, (req, res, next) => {
-  const message = req.body;
+  const message = req.body.message;
```
Comment (matan-co) [inline server/api/v1/message/message.js:11]: Please read on javascript destruction . 
https://dmitripavlutin.com/javascript-object-destructuring/

Comment (matan-co) [reply server/api/v1/message/message.js:11]: image

Comment (matan-co) [reply server/api/v1/message/message.js:11]: you can do const {message} = req.body;

**Review by matan-co — CHANGES_REQUESTED** [review]
Nice job 😁👌
Added some comments

File: client/public/index.js (hunk 4 above)
Comment (kobiez) [reply client/public/index.js:4]: I ran into an issue when trying to split as you asked, I wasn't able to export the userId variable from authService file to messageService file.

File: client/public/index.html (hunk 2 above)
Comment (kobiez) [reply client/public/index.html:14]: Moved it

File: client/public/index.js (hunk 3 above)
Comment (kobiez) [reply client/public/index.js:66]: Done

File: server/api/v1/auth/auth.test.js (hunk 5 above)
Comment (kobiez) [reply server/api/v1/auth/auth.test.js:8]: Done

File: server/api/v1/auth/auth.test.js (hunk 6 above)
Comment (kobiez) [reply server/api/v1/auth/auth.test.js:10]: Done

File: server/api/v1/message/message.js (hunk 7 above)
Comment (kobiez) [reply server/api/v1/message/message.js:11]: Fixed it. 
Read it.

**Review by matan-co — APPROVED** [review]

**Pull request merged**